crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -1962,6 +1962,22 @@ impl<'src> Parser<'src> {
         helpers::set_expr_ctx(&mut target, ExprContext::Store);
 
         let value = self.parse_conditional_expression_or_higher();
+        if !value.is_parenthesized {
+            match value.expr {
+                Expr::Yield(_) | Expr::YieldFrom(_) => {
+                    self.add_error(
+                        ParseErrorType::OtherError(
+                            "unparenthesized yield expression cannot be used here".to_string(),
+                        ),
+                        &value,
+                    );
+                }
+                Expr::Starred(_) => {
+                    self.add_error(ParseErrorType::StarredExpressionUsage, &value);
+                }
+                _ => {}
+            }
+        }
```

Comment:
Is this part of this PR or is this here due to some rebasing?

Reply:
Yes, this is part of this PR. It's to disallow certain expressions on the right-side (`x := <here>`).